### ISSUE TITLE
Do you plan on releasing the training code?

### ISSUE THREAD

**Issue by yinoue0426** [LouisNUST/CrackFormer-II#2]
Hello,

I am amazed by your work. If possible, I would like to know whether you plan on releasing the training code so I can use it for my project.
If not, I don't mind re-implementing the training code myself, but I have a few questions about few training parameters:

First, according to your paper, the prediction is discretized to 3 classes during the loss calculation: negative if the predicted probability is less than 0.05, positive if the predicted probability is larger than η, and neutral if the predicted probability is between 0.05 and η. The loss is not calculated for the neutral class. What value is used for η? Is it 0.95, since 1-0.05=0.95?

Second, as I understand, the loss is calculated by combining the loss values for all side outputs as well as the fused output. The summation is weighted by the loss weight S^k_{side} and S_{fuse}, and they are all different (in other words, there are 6 loss weights). What values did you choose for them?

Third, the paper states that clipping, flipping, rotation, and gamma transformation are applied as augmentations. However, the data loader class (https://github.com/LouisNUST/CrackFormer-II/blob/main/CrackFormer-II/utils/Crackloader.py) does not seem to contain any augmentations, so I'd like to know the parameters for each transformations.
Also, since the paper says "In the end, we expand each training set by 228 times of the original samples." I assume that the augmentation was performed offline. The paper also says that you trained the network for 500 epochs. In your setting, is 1 epoch counted as going over every image in the 228 times augmented dataset? Or does that count as 228 epochs?

Finally, the paper says that the model is trained with a StepLR learning rate scheduler. However, that does not seem to be the case according to the code (https://github.com/LouisNUST/CrackFormer-II/blob/871a3cdb9a473095be78faee487e4b78bbce0d3a/CrackFormer-II/utils/utils.py). Which is true?

Sorry for the long post, if it takes a long time to prepare the code, I am ok if you just reply to the questions.
Best,

**Comment by jo1boy** [comment]
Hi，I'm learning this work,too. But I don't have enough ability to re-implementing the training code.So if you already finished the training code, please share with me. Thanks a lot!

**Comment by viokula** [comment]
> Hi，I'm learning this work,too. But I don't have enough ability to re-implementing the training code.So if you already finished the training code, please share with me. Thanks a lot!

do you reproduce the performance that this paper achieve ? I can't reproduce the result in my training code. So if you reproduce the result, please share with me.
